Build failure: fcitx5-qt6-5.1.10

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

build the package on unstable small

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/310479366

### Relevant log output

```console
CMake Error at qt6/platforminputcontext/CMakeLists.txt:43 (target_link_libraries):
  Target "fcitx5platforminputcontextplugin-qt6" links to:

    Qt6::GuiPrivate

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```

### Additional context

_No response_

### System metadata

fetching path input 'path:/nix/store/sjc22xsglvk0j87snq2mqfa42vjzrlvj-source'
 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.3, NixOS, 25.11 (Xantusia), 25.11.20251020.40ef6b9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: ~/.config/nix/nix.conf:/nix/store/18nfhj7k76g7pdva8fp5xdxda6158kyg-stylix-kde-config/nix/nix.conf:/etc/xdg/nix/nix.conf:~/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:~/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/<username>/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/968c8420jvmpwpi0rhqxw2938svw25mi-lix-2.93.3/share`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/nix/store/sjc22xsglvk0j87snq2mqfa42vjzrlvj-source`

### Notify maintainers

    @poscat0x04

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/inputmethods/fcitx5/fcitx5-qt.nix
+++ b/pkgs/tools/inputmethods/fcitx5/fcitx5-qt.nix
@@ -10,6 +10,7 @@
   qtwayland,
   wrapQtAppsHook,
   wayland,
+  fetchpatch2,
 }:
 let
   majorVersion = lib.versions.major qtbase.version;
@@ -24,6 +25,22 @@ stdenv.mkDerivation rec {
     rev = version;
     hash = "sha256-JhmaAAJ1fevCPItVnneUCAalnDDaCjjkAl9QRhSkBk4=";
   };
+
+  patches = [
+    # TODO: remove on next release
+    (fetchpatch2 {
+      url = "https://github.com/fcitx/fcitx5-qt/commit/46a07a85d191fd77a1efc39c8ed43d0cd87788d2.patch?full_index=1";
+      hash = "sha256-qv8Rj6YoFdMQLOB2R9LGgwCHKdhEji0Sg67W37jSIac=";
+    })
+    (fetchpatch2 {
+      url = "https://github.com/fcitx/fcitx5-qt/commit/6ac4fdd8e90ff9c25a5219e15e83740fa38c9c71.patch?full_index=1";
+      hash = "sha256-x0OdlIVmwVuq2TfBlgmfwaQszXLxwRFVf+gEU224uVA=";
+    })
+    (fetchpatch2 {
+      url = "https://github.com/fcitx/fcitx5-qt/commit/1d07f7e8d6a7ae8651eda658f87ab0c9df08bef4.patch?full_index=1";
+      hash = "sha256-22tKD7sbsTJcNqur9/Uf+XAvMvA7tzNQ9hUCMm+E+E0=";
+    })
+  ];
 
   postPatch = ''
     substituteInPlace qt${majorVersion}/platforminputcontext/CMakeLists.txt \

